scipy/optimize/_minpack_py.py

```diff
@@ -624,7 +625,8 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         If True, check that the input arrays do not contain nans of infs,
         and raise a ValueError if they do. Setting this parameter to
         False may silently produce nonsensical results if the input arrays
-        do contain nans. Default is True.
+        do contain nans. Default is True. Note that if `nan_policy` is
+        specified explicitly (not None), this value will be ignored.
```

Comment:
This is no longer true, actually. It's really that if `nan_policy` is specified explicitly and `check_finite` is _not_ specified specifically, then `check_finite` is ignored - or really, the default becomes False.

But let's fix this in a followup. What we really should do is deprecate `check_finite` and have a new `inf_policy` so that the `inf` and `nan` checks are totally independent.

Reply:
Resolved by gh-17852.